src/FluffySpoon.AspNet.LetsEncrypt.Aws/IAwsDnsChallengePersistenceStrategy.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading.Tasks;
+
+namespace FluffySpoon.AspNet.LetsEncrypt.Aws
+{
+	public interface IAwsDnsChallengePersistenceStrategy
```

Comment:
Ordinarily I wouldn't, but I was following your existing code. The Azure persistence challenge has its own interface, despite it being a concrete implementation. I'd quite happily remove it.

Reply:
It is necessary to know what the record type is. Requiring that every DNS strategy split the key to figure out what it means based on some arbitrary format seems a bit backwards. It is also necessary to be able to determine which strategies are specifically intended for DNS and which are for HTTP (due to the way the ACME protocol operates), so some differentiating interface would be necessary regardless.

Reply:
Requiring all DNS strategies to decode the key to figure out what record type is required seems a bit backwards, and would go against your policy of making the code as self-describing as possible. It is also necessary to be able to separate the HTTP strategies from the DNS strategies so that challenges can be routed appropriately. Also, DNS strategies need to be able to remove records after they have been verified, which is not something that the existing interface allows. Adding that to the existing interface would be a breaking change.

Reply:
Actually, just looked at the existing interface and it doesn't allow a string key - it's just the PersistenceType enum and a byte array to persist.

Reply:
Alright - perfect. As long as we can update the documentation to reflect these changes, and keep it understandable. Code is only as good as its documentation.

Reply:
I've already updated the README with basic instructions that hopefully make it clear that a different interface must be used. I'll review it again and may add some more.